src/board.rs

```diff
@@ -762,6 +780,40 @@ impl Board {
                 return true;
             }
         }
+
+        unsafe {
+            // A new game has started
+            if self.move_history.is_empty() {
+                BOARD_HISTORY.clear();
+                BOARD_HISTORY.push(self.board);
+                return false;
+            }
+        }
+
+        unsafe {
+            // Add the new move
+            BOARD_HISTORY.push(self.board);
+
+            // Index mapping
+            let num_positions: usize = BOARD_HISTORY.len();
+            let mut count_equal = vec![1; num_positions];
+            #[allow(clippy::needless_range_loop)]
+            for i in 0..(num_positions - 1) {
+                for j in (i + 1)..num_positions {
+                    if equal_boards(BOARD_HISTORY[i], BOARD_HISTORY[j]) {
+                        count_equal[i] += 1;
+                    }
+                }
+            }
+
+            // Checking for unclaimed draw by threefold repetition
+            for b in count_equal.iter() {
+                if *b >= 3 {
+                    return true;
+                }
+            }
+        }
+
```

Comment:
Maybe something like that works 
```rust
// Start a new game if the move history is empty
if self.move_history.is_empty() {
    self.board_history.clear();      // Clear history
    self.board_history.push(self.board); // Add initial board state
    return false;
}

// Add the current board state to history
self.board_history.push(self.board);

// Count occurrences of each board position
let mut position_counts = std::collections::HashMap::new();
for board in self.board_history.iter() {
    let count = position_counts.entry(board).or_insert(0);
    *count += 1;
}

// Check for threefold repetition
if position_counts.values().any(|&count| count >= 3) {
    return true;
}

false
```

Reply:
Sure, I will rewrite it (I think you suggest to make a variable in the Board class) and add also some optimizations (ex check if count is >= 3 when increasing it...), but the idea is the same. I will send another PR tomorrow with this updates.

Reply:
You can just commit on top of this pr, the most important thing is to be able to remove the unsafe block

Reply:
I will be glad if you want to add something else (or request) besides this, to know where I have to improve the code and minimize the number of PRs.

Reply:
It seems it worked with the board_history declared int the Board structure.

Reply:
Hey @TomaDumitrescu thanks for the modification, I saw that there is the `draw_by_repetition` test, can you modify it before I can merge

Reply:
What do you mean by modifying draw_by_repetition test?

Reply:
I saw what the problem was. In a game simulation, is_draw() is called after every move (this is how I based my logic on creating this function). However, for consecutive_positions_draw, onoy move_history was tested, and there was no call of draw_by_repetition() to update the moves in the board_history... I did also this.

Reply:
Also, I don't have clippy errors

Reply:
I will take a look, although it is strange because I don't have bugs. Maybe the wrong version loaded (there is no derive at line 15... That was from an old version...)

Reply:
I think it's something unfair here... I think you can delete the line with impl Eq for PieceType {} and just let the derive. That line wasn't there when I cloned the repository, and also it makes more sense to derive the Eq.

Reply:
So, THE BUG IS IN THE PTEVIOUS MERGED PR, NOT ON THIS

Reply:
What's your opinion on this?

Reply:
I tested the solution it seems to work, I like the way you store the entire board each time, this might be useful for the #67 issue, checking if the code github action is working then merging, thanks again for your contribution 🎉

Reply:
Why don't you merge the PR? It's MY idea... Also, it works. The bug you are seeing is that you can't have Eq on Derive and also impl Eq for... (there are 2 def, one of them should be deleted). I DON'T want someone else to solve this, just to let others to contribute easily without working. I spent too much time, and I consider it unfair!

Reply:
Your fix with eq on derive will create 2 definitions of Eq for PieceType (AS you might already knkw, but perhaps want to dismiss my solution incorrectly...). Either you delete impl Eq for PieceType {} at line ... on mod.rs from /pieces and put Eq in Derive as I suggested (Which I cannot do, because I don't have permissions), either you let THIS MERGE, that will have THE EQ implemented WITHIN your project, ALTHOUGH it is not on mine, because on HOW GITHUB MERGE WORKS.

Reply:
@thomas-mauran Did you read my previous message regarding the update? It's true that when you clone this will not work, because there is no Eq implemented. However, if you add one, it will conflict with the already existing Eq implmentation in chess tui project

Reply:
Ok, you merged some other PRs in the meantime and you implemented some things for PieceType that now lead to some errors because I don't have access to the updated code. I hope this contribution will not be taken by someone else.

Reply:
Your code currently doesn't run when running a cargo run and on commit [ec8716b](https://github.com/thomas-mauran/chess-tui/pull/82/commits/ec8716b67d5562ffb6fe630bd02950e1d130d26d) the cargo run didn't work either so I couldn't merge it on main. 

The modifications pushed on the main branch are not affecting your branch since there was no rebase on it, neither merges. Don't worry about anybody stealing this PR from you, it will obviously not happen since this is affected to you.

It took me a while to understand what you were talking about with all of those messages and PR but I think I see what you mean with the impl Eq from the last merged pr. What you NEED to do is to rebase main on your branch to have this code and make this pr work.

I cannot merge a PR if either the CI lint or cargo run don't work, I need to see it working on this branch first.

Reply:
Good to know! Thank you. Now, I removed all files, cloned chess_tui, modified, tested with github workflows, and then pushed on GitHub. It should work now (because locally it works...).

Reply:
Good, you only need to resolve the conflict on the brannch now. Else you could have used git rebase to apply the main branch commit on top of yours instead of this commit [3033374](https://github.com/thomas-mauran/chess-tui/pull/82/commits/30333744628a6f1984788c0e7a52cf507f002e71) I think it would have sorted out the issues. @ me if you need help with the conflict

Reply:
Now it should work